census/vanilla-tf/trainer/task.py

```diff
@@ -0,0 +1,335 @@
+import os
+
+tf.logging.set_verbosity(tf.logging.INFO)
+
+#See tutorial on wide and deep https://www.tensorflow.org/tutorials/wide_and_deep/
+#https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/feature_column.py
+
+# csv columns in the input file
+CSV_COLUMNS = ('age', 'workclass', 'fnlwgt', 'education', 'education_num',
+               'marital_status', 'occupation', 'relationship', 'race', 'gender',
+               'capital_gain', 'capital_loss', 'hours_per_week', 'native_country',
+               'income_bracket')
+
+CATEGORICAL_COLS = ('gender', 'race', 'education', 'marital_status',
+                    'relationship', 'workclass','occupation', 'native_country')
+
+CONTINUOUS_COLS = ('age', 'education_num', 'capital_gain', 'capital_loss', 'hours_per_week')
+
+LABEL_COL = 'income_bracket'
+
+
+#
+# Graph creation section
+# The three functions below create everything for the graph.
+#
+def loss(logits, labels):
+  """Compute cross entropy loss function."""
+  cross_entropy = tf.reduce_mean(
+      tf.nn.softmax_cross_entropy_with_logits(logits=logits, labels=labels))
+  return cross_entropy
+
+def optimizer(loss, global_step):
+  """Gradient descent optimizer with 0.5 learning rate."""
+  train_step = tf.train.GradientDescentOptimizer(0.5).minimize(
+      loss, global_step=global_step)
+  return train_step
+
+def evaluation_step(logits, labels):
+  """Compute the graph for the accuracy step."""
+  correct_prediction = tf.equal(tf.argmax(logits, 1), tf.argmax(labels, 1))
+  accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
+  return accuracy
+
+#
+# Read the input and process
+#
+def get_inputs_and_labels():
+  """Placeholder for inputs and labels
+     input shape = [None, 337] label shape = [None, 2]."""
+  inputs = tf.placeholder(tf.float32, shape=[None, 337])
+  labels = tf.placeholder(tf.float32, shape=[None, 2])
+  return (inputs, labels)
+
+def read_input_data(file_name, skiprows=None):
+  """Read the input data as a pandas DataFrame of features and labels."""
+  input_df = pd.read_csv(StringIO(tf.gfile.GFile(file_name).read()), names=CSV_COLUMNS, skiprows=skiprows)
+
+  label_df = input_df.pop(LABEL_COL)
+  return (input_df, label_df)
+
+def generate_input(input_df, label_df):
+  """Prepare the input columns using SparseTensor."""
+
+  # convert the continuous columns into tf.constant tensor
+  continuous_columns = [
+      tf.constant(input_df[col].values) for col in CONTINUOUS_COLS
+  ]
+
+  # convert the categorical columns into sparse tensors
+  categorical_columns = [
+      tf.SparseTensor(
+          indices=[[i, 0] for i in range(input_df[col].size)],
+          values=input_df[col].astype('category').cat.codes.values,
+          dense_shape=[input_df[col].size, 1])
+      for col in CATEGORICAL_COLS
+  ]
+
+  # convert the labels into one hot encoding
+  label_tensor = tf.one_hot(
+      label_df.astype('category').cat.codes.values,
+      2, off_value=1, on_value=0)
```

Comment:
I guess I was trying to convert 1 to [1 0] but not needed. Getting rid of it. Done.